uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -130,6 +128,27 @@ protected function update($params) {
     return $this->createRequestWithBalanceChanges($params);
   }
 
+  /**
+   * Return an error message
+   * either for specific invalid status transition cases
+   * or a default generic message
+   *
+   * @param array $params
+   *
+   * @return string
+   */
+  private function getErrorMessageForInvalidStatusTransition($params) {
+    $leaveStatuses = LeaveRequest::getStatuses();
+    $isOwnRequest = CRM_Core_Session::getLoggedInContactID() === $params['contact_id'];
```

Comment:
Please use the `$this->leaveRequestRightsService->currentUserIsLeaveContact` method here

Reply:
Done.